iohk/Appveyor.hs

```diff
@@ -0,0 +1,79 @@
+{-# OPTIONS_GHC -Weverything -Wno-unsafe -Wno-implicit-prelude -Wno-semigroup #-}
+{-# LANGUAGE DeriveGeneric              #-}
+{-# LANGUAGE FlexibleInstances          #-}
+{-# LANGUAGE FunctionalDependencies     #-}
+{-# LANGUAGE GeneralizedNewtypeDeriving #-}
+{-# LANGUAGE OverloadedStrings          #-}
+{-# LANGUAGE TemplateHaskell            #-}
+{-# LANGUAGE DataKinds                  #-}
+
+module Appveyor where
+
+import           Control.Lens.TH (makeFields)
+import           Data.Aeson      (FromJSON, parseJSON, withObject, (.:))
+import           Data.Monoid     ((<>))
+import           Data.String     (IsString)
+import qualified Data.Text       as T
+import           GHC.Generics    (Generic)
+import           Utils           (fetchJson)
+import           Types           (ApplicationVersion(ApplicationVersion), Arch(Win64))
+import           Data.Coerce                      (coerce)
+
+newtype JobId = JobId T.Text deriving (Show, Monoid)
+newtype BuildNumber = BuildNumber Integer deriving (Show)
+newtype Username = Username { usernameToText :: T.Text } deriving (Show, Monoid, IsString)
+newtype Project = Project { projectToText :: T.Text } deriving (Show, Monoid, IsString)
+
+data AppveyorBuild = AppveyorBuild {
+    _appveyorBuildBuild :: AppveyorBuild2
+    } deriving (Show, Generic)
+
+data AppveyorBuild2 = AppveyorBuild2 {
+    _appveyorBuild2Jobs         :: [AppveyorBuild3]
+    ,_appveyorBuild2BuildNumber :: BuildNumber
+    ,_appveyorBuild2Version     :: ApplicationVersion Win64
+    } deriving (Show, Generic)
+
+data AppveyorBuild3 = AppveyorBuild3 {
+    _appveyorBuild3JobId :: JobId
+    } deriving (Show, Generic)
+
+data AppveyorArtifact = AppveyorArtifact {
+    fileName :: T.Text
+    , name   :: T.Text
+    } deriving (Show, Generic)
+
+makeFields ''AppveyorBuild
+makeFields ''AppveyorBuild2
+makeFields ''AppveyorBuild3
+makeFields ''AppveyorArtifact
+
+instance FromJSON AppveyorBuild where
```

Comment:
Are we allowed the use of `TemplateHaskell` for such scripts?

In theory if we do `deriveFromJSON defaultOptions { fieldLabelModifier = drop 5 }` we should be in business, shouldn't we?

Reply:
i was running into problems where many different objects had fields with the same names, and i wanted to try using a lens to access them